### PR TITLE
PDTP fixes

### PR DESCRIPTION
**Pull request by abigailgold**
# Description

Fix probability check for multi-dimensional arrays + out of bounds error in binning in PDTP metric implementation

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [X] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing
All unit tests pass

**Test Configuration**:
- OS: MacOS 12.5
- Python version: 3.7
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abigailgold** [comment]
New PR #1825 was opened instead of this one.
